files/default/neutron-ha-tool.py

```diff
@@ -233,6 +236,10 @@ def run(args):
     else:
         raise ValueError('Invalid agent_selection_mode')
 
+    if args.target_agent is not None:
+        Configuration.agent_picker_class = SingleAgentPicker
+        SingleAgentPicker.agent_selection_value = args.target_agent
```

Comment:
uhm this is a bit ugly, this value is now for all SingleAgentPicker instances, because you are updating the class

But TBH I do not see a better solution if we work with classes instead of instances. 

Maybe a better solution is to pass the strategy into `migrate_l3_routers_from_agent` via `l3_agent_evacuate` and `l3_agent_migrate` as a parameter. But I agree that this will require more refactoring : (((